### PR TITLE
WIP: Theme support

### PR DESCRIPTION
**Pull request by justinline**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   preset: "ts-jest",
   moduleNameMapper: {
+    "@/(.*)$": "<rootDir>/src/$1",
```
Comment (justinline) [inline jest.config.js:4]: necessary cause I used `@/` as a module import in a tested file

File: src/App/App.css [hunk 2]
```diff
@@ -7,5 +7,8 @@
   margin-left: 1rem;
   margin-top: 0.8rem;
   z-index: 1;
-  background-image: url("resources/feather/menu.svg");
+  background-color: var(--task-color);
+  mask-size: contain;
```
Comment (justinline) [inline src/App/App.css:11]: need to test if this works on browsers other than firefox

**Comment by justinline** [comment]
@etrobert feel free to check this out and see what you think and whether you'd consider it as a feature request :)

File: src/theme/ThemeSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const ThemeSwitcher = () => {
+  const { setTheme } = useContext(ThemeContext);
+
+  return (
+    <ul style={{ display: "flex", justifyContent: "space-between" }}>
+      {themes.map((item) => (
+        <li key={item.title}>
+          <button
+            style={{
```
Comment (justinline) [inline src/theme/ThemeSwitcher.tsx:14]: chore: put this into a class

File: src/App/MenuBar/MenuBar.tsx [hunk 4]
```diff
@@ -38,7 +46,7 @@ const MenuBar = ({ open, onClose, onNewGraph }: Props): JSX.Element => {
         target="_blank"
       />
       <p>
-        Made by
+        Made by{" "}
```
Comment (justinline) [inline src/App/MenuBar/MenuBar.tsx:49]: fixed a missing space 😇

**Comment by etrobert** [comment]
It is pretty cool and interesting!

But I don't know if having the whole app with a custom theme is something desirable tbh...

It would make it harder to have a TaskGraph branding

Plus that's a lot of code and complexity to maintain for not a lot of benefit for the user

I can't even edit a Task name right now 😅 

I guess a Dark/Light version could be desirable but in this case I would rather not do any computation in JS and have the CSS handle the theming using `color-scheme`...

Source: https://www.youtube.com/watch?v=oHcTn83M1ls&list=PLNYkxOF6rcIAaV1wwI9540OC_3XoIzMjQ

I'm sorry I feel like a dick, you've worked hard on this but I'm not sure this is desirable in the first place 😭

**Comment by justinline** [comment]
haha, it's totally fine, I didn't work so hard on it - it took like 1 hour max and was just a prototype

I disagree with "Not a lot of benefit for the user" I pretty much avoid using apps without dark mode these days, so i'll continue to not use taskgraph until that's implemented at a later date 😛 

The prefers color-scheme is a perfectly viable route also if it's just going to be dark/light - this code could easily be adapted by deleting the provider and putting it into css

can revisit it at a later date 😇

**Comment by etrobert** [comment]
> I also think you should open-source your ideas/priorities and roadmaps as gh issues so I can contribute effectively on things you do/don't want 🤗 

This is absolutely right and actually I have this managed in a Notion page that I will make public!

**Comment by etrobert** [comment]
> I also think you should open-source your ideas/priorities and roadmaps as gh issues so I can contribute effectively on things you do/don't want 🤗

Actually I just realised that the ideas/priorities are already linked in the project README!

There is a link to the project's Notion page.

Then you can go to any of the 4 project parts `Code | Design | Product | Project` and you will see a current list of tasks related to this part!

**Pull request closed without merging**